src/main/java/seedu/address/model/session/SessionDate.java

```diff
@@ -144,6 +151,17 @@ public static boolean isValidSessionDate(String dateValue, String timeValue) {
         }
     }
 
+    //does a 2038 problem check on year after parsing
+    private static boolean isValidYear(String dateValue) {
```

Comment:
Hi, actually not sure where this bug occurs, but when I was testing I noticed there is a weird bug now. As seen below, I added a 1970 session to TutorBuddy, but then in sessions, it displays the date is 1971 instead. Its the same for other values where it +1 to the session. Storage still shows the correct value, so not sure if the +1 only comes in in the UI. Seems to happen in master also, u might want to give it a try.

![image](https://user-images.githubusercontent.com/24240408/114161253-6f661700-995a-11eb-9344-cda3d63b11d0.png)

Reply:
Good catch, I tried on the master branch this seems to be a problem from there